drivers/abp2/abp2.c

```diff
@@ -0,0 +1,242 @@
+#define ABP2_CMD_NOP        (0xF0)          /**< NOP command. */
+#define ABP2_RX_BUF_LEN     (8)             /**< Length of the receive buffer. */
+
+/** Data to send in order to start a measurement. */
+static const uint8_t data_tx_meas_start[] = { ABP2_CMD_MEAS, 0x00, 0x00 };
+
+/** Data to send in order to start a measurement and read the data of the previous measurement. */
+static const uint8_t data_tx_meas_read[] = { ABP2_CMD_MEAS, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 };
+
+/** Data to send in order to read the data of the previous measurement. */
+static const uint8_t data_tx_nop_read[] = { ABP2_CMD_NOP, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 };
+
+static const int32_t cntPMax = 15099494;    /**< Output at maximum pressure (counts). */
+static const int32_t cntPMin = 1677722;     /**< Output at minimum pressure (counts). */
+
+int abp2_init(abp2_t *dev, const abp2_params_t *params)
+{
+    uint8_t status = 0xFF;      /* sensor status byte */
+    int res = 0;
+    int count = 10;
+
+    assert(dev && params);
+    dev->params = params;
+
+#if defined(MODULE_ABP2_SPI)
+    res = spi_init_cs(params->spi, params->cs);
+#else
+#pragma message("implement I2C code here")
+#endif
+    if (res) {
+        return -EIO;
+    }
+
+    /* test status byte: the busy flag should clear in about 5ms */
+    while ((status & ABP2_STATUS_BUSY) && count) {
+#if defined(MODULE_ABP2_SPI)
+        spi_acquire(params->spi, params->cs, SPI_MODE_0, params->clk);
+        status = spi_transfer_byte(params->spi, params->cs, false, ABP2_CMD_NOP);
+        spi_release(params->spi);
+#else
+#pragma message("implement I2C code here")
+#endif
+        ztimer_sleep(ZTIMER_MSEC, 1);
+        count--;
+    }
+    /* the busy flag should be clear */
+    if (status & ABP2_STATUS_BUSY) {
+        res = -ETIMEDOUT;
+    }
+
+    return res;
+}
+
+int abp2_measure(const abp2_t *dev)
+{
+    uint8_t status = 0xFF;      /* sensor status byte */
+    int res = 0;
+    uint8_t data_rx[ABP2_RX_BUF_LEN];
+
+#if defined(MODULE_ABP2_SPI)
+    spi_acquire(dev->params->spi, dev->params->cs, SPI_MODE_0, dev->params->clk);
+    spi_transfer_bytes(dev->params->spi, dev->params->cs, false, data_tx_meas_start, data_rx,
+                       sizeof(data_tx_meas_start));
+    spi_release(dev->params->spi);
+#else
+#pragma message("implement I2C code here")
+#endif
+    status = data_rx[0];
+    if (status & ABP2_STATUS_MASK) {
+        return -ENODATA;
+    }
+
+    return res;
+}
+
+int abp2_read(const abp2_t *dev, int32_t *press, int32_t *temp)
+{
+    int res = 0;
+    uint8_t status = 0xFF;      /* sensor status byte */
+    int loopsTimeout = MAX_LOOPS_TIMEOUT;
+    abp2_raw_t rawData;
+
+    res = abp2_measure(dev);
+    if (res) {
+        return res;
+    }
+    /* wait until the busy flag clears or until timeout: */
+    while (((status = abp2_getstatus(dev)) & ABP2_STATUS_BUSY) && loopsTimeout) {
+        loopsTimeout--;
+        ztimer_sleep(ZTIMER_MSEC, 1);   /* wait 1ms */
```

Comment:
short: single digit sleep is not reliable (use usec or if you know a little more about the conversion time use that value ( seems like the conversion time is ~ 5ms) and a busy wait )

why: every timer has a jitter of at least 1 due to it might be set just after or just before the count change 

if the count isn't actually used but scaled an the used it might seem reliable until someone changes that scaling.